domains.js

```diff
@@ -0,0 +1,72 @@
+const google = require('googleapis');
+const sheets = google.sheets('v4');
+
+function getDomains(username) {
+    const credentials = credentialConfiguration();
+    const request = {
+        spreadsheetId: credentials.TASK_SHEET_ID,
+        range: 'A:ZZZ', // extreme range to get whole spreadsheet
+        auth: credentials.API_KEY
+    };
+
+    return new Promise((resolve, reject) => {
+        sheets.spreadsheets.values.get(request, function(err, response) {
+            if (err) {
+                reject(err);
+            } else {
+                const domains = findMatch(username, response.values);
+                if (domains) {
+                    resolve({ 'domains': domains })
+                } else {
+                    reject({ 'error': `${username} not found.`})
+                }
+            };
+        });
+    });
+}
+
+function findMatch(username, records) {
+    let domains = records.filter(record => {
+        return username.toLowerCase() === record[0].toLowerCase()
+    })[0];
+
+    if (domains) {
+        return domains.slice(1);
+    } else {
+        return null;
+    }
+}
+
+let baseEnvironment;
+
+/**
+ * Create a credential object with secret values from process.env.
+ * In development, also read from a file named .env, if present.
+ * @returns {Object}
+ */
+function credentialConfiguration () {
```

Comment:
**A1**: Don't want to worry about config stuff right now and everything in this will be deleted, but I will remove the unnecessary filtering. Can we ticket config module for later?